.pre-commit-config.yaml

```diff
@@ -54,3 +54,15 @@ repos:
         types_or: [c++, c, cuda]
         exclude: |
           (?x)^(3rdparty/.* flashinfer/jit/aot_config.py)$
+
+  -   repo: https://github.com/pre-commit/mirrors-mypy
+      rev: ''  # Use the sha / tag you want to point at
+      hooks:
+      -   id: mypy
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    # Ruff version.
+    rev: v0.12.7
+    hooks:
+      # Run the linter.
+      - id: ruff-check
```

Comment:
also add
```
    - id: ruff-format
      types_or: [ python, pyi ]
```

and add an empty new line in the end.

Reply:
added and formatter